examples/plot_callosal_tract_profile.py

```diff
@@ -0,0 +1,558 @@
+  callosum templates:
+
+    https://figshare.com/articles/Templates_for_Automated_Fiber_Quantification_of_corpus_callosum_from_Diffusion_MRI_data/3381523
+
+"""
+import os.path as op
+import matplotlib.pyplot as plt
+import numpy as np
+import nibabel as nib
+import dipy.data as dpd
+from dipy.data import fetcher
+from dipy.io.streamline import save_tractogram, load_tractogram
+from dipy.stats.analysis import afq_profile, gaussian_weights
+from dipy.io.stateful_tractogram import StatefulTractogram
+from dipy.io.stateful_tractogram import Space
+
+import AFQ.data as afd
+import AFQ.tractography as aft
+import AFQ.registration as reg
+import AFQ.models.dti as dti
+import AFQ.segmentation as seg
+from AFQ.utils.volume import patch_up_roi
+
+import logging
+import sys
+
+# Ensure segmentation logging information is included in this example's output
+root = logging.getLogger()
+root.setLevel(logging.ERROR)
+logging.getLogger('AFQ.Segmentation').setLevel(logging.INFO)
+logging.getLogger('AFQ.tractography').setLevel(logging.INFO)
+handler = logging.StreamHandler(sys.stdout)
+handler.setLevel(logging.INFO)
+root.addHandler(handler)
+
+# Monkey patch the progress bars so don't pollute this example's output
+import AFQ._fixes as fixes
+
+def notqdm(iterable, *args, **kwargs):
+    """
+    replacement for tqdm that just passes back the iterable to silence `tqdm`
+    """
+    return iterable
+
+
+afd.tqdm = notqdm
+fixes.tqdm = notqdm
+seg.tqdm = notqdm
+
+
+def update_progressbar(progress, total_length):
+    return
+
+
+fetcher.update_progressbar = update_progressbar
+
+
+# Target directory for this example's output files
+working_dir = "./callosal_tract_profile"
+
+##########################################################################
+# Helper functions:
+# -----------------
+
+
+def show_anatomical_slices(img_data, title):
+    """
+    display anatomical slices from midpoint
+
+    based on:
+    https://nipy.org/nibabel/coordinate_systems.html
+    """
+
+    axial_slice = img_data[:, :, int(img_data.shape[2] / 2)]
+    coronal_slice = img_data[:, int(img_data.shape[1] / 2), :]
+    sagittal_slice = img_data[int(img_data.shape[0] / 2), :, :]
+
+    fig = plt.figure(constrained_layout=False)
+    gs = fig.add_gridspec(nrows=3, ncols=2, wspace=0.01, hspace=0.01)
+    ax1 = fig.add_subplot(gs[:-1, :])
+    ax1.imshow(axial_slice.T, cmap="gray", origin="lower")
+    ax1.axis('off')
+    ax2 = fig.add_subplot(gs[2, 0])
+    ax2.imshow(coronal_slice.T, cmap="gray", origin="lower")
+    ax2.axis('off')
+    ax3 = fig.add_subplot(gs[2, 1])
+    ax3.imshow(sagittal_slice.T, cmap="gray", origin="lower")
+    ax3.axis('off')
+
+    plt.suptitle(title)
+    plt.show()
+
+# Imports for create_density_map
+
+
+from dipy.io.utils import (create_nifti_header, get_reference_info)
+from dipy.tracking.streamline import select_random_set_of_streamlines
+from dipy.tracking.utils import density_map
+
+
+def create_density_map(tractogram, fname):
```

Comment:
Consider streamline density map diff. This was one of the first considerations I had when review output. In discussion with the group seems like we could go multiple directions. Here's summary:

from @arokem 

> The main use for visitation maps or streamline density maps is in measuring the volume of certain tracts (i.e., the number of voxels visited) and in comparing two segmentations of the same brain to each other.

One approach is to use Dice's coefficient
> We would use the two visitation maps to do calculate a Dice coefficient between the two measurements/segmentations, which gives as an indication of the test-retest reliability of the segmentation in this case.

Other approaches worth mentioning that address spirit of streamline differences
- leveraging return idx and comparing (from @36000)
- comparing tract profiles (from @36000)
- utilizing convex hull comparison (from @richford)
- alpha shapes in plotly (from @richford)

Reply:
See #517